homeassistant/components/jellyfin/media_player.py

```diff
@@ -0,0 +1,489 @@
+            action="POST",
+            params={"playCommand": "PlayNow", "itemIds": [media_id]},
+        )
+
+    def set_volume_level(self, volume: float) -> None:
+        """Set volume level, range 0..1."""
+        self._client.jellyfin.sessions(
+            handler=f"/{self._id}/Command",
+            action="POST",
+            json={"Name": "SetVolume", "Arguments": {"Volume": int(volume * 100)}},
+        )
+
+    def mute_volume(self, mute: bool) -> None:
+        """Mute the volume."""
+        self._client.jellyfin.sessions(
+            handler=f"/{self._id}/Command",
+            action="POST",
+            json={"Name": "Mute" if mute else "Unmute"},
+        )
+
+    @property
+    def device_info(self) -> DeviceInfo | None:
+        """Return a device description for device registry."""
+
+        if not self._data["Capabilities"]["SupportsPersistentIdentifier"]:
+            return DeviceInfo(
+                identifiers={(DOMAIN, "Jellyfin-clients")},
+                name="Jellyfin Client Service",
+                manufacturer="Jellyfin",
+                model="Jellyfin Clients",
+                entry_type=DeviceEntryType.SERVICE,
+            )
+
+        return DeviceInfo(
+            identifiers={(DOMAIN, self._data["DeviceId"])},
+            manufacturer="Jellyfin",
+            model=self._data["Client"],
+            name=self.name,
+            sw_version=self._data["ApplicationVersion"],
+            via_device=(DOMAIN, self._data["ServerId"]),
+        )
+
+    async def async_browse_media(
+        self, media_content_type: str | None = None, media_content_id: str | None = None
+    ) -> BrowseMedia:
+        """Return a BrowseMedia instance.
+
+        The BrowseMedia instance will be used by the "media_player/browse_media" websocket command.
+
+        """
+
+        def browse_media_from_item(media: Any) -> BrowseMedia:
+            content_type_map = {
+                "Series": MEDIA_TYPE_TVSHOW,
+                "Movie": MEDIA_TYPE_MOVIE,
+                "CollectionFolder": "collection",
+                "Folder": "library",
+                "BoxSet": "boxset",
+            }
+            media_class_map = {
+                "Series": MEDIA_CLASS_DIRECTORY,
+                "Movie": MEDIA_CLASS_MOVIE,
+                "CollectionFolder": MEDIA_CLASS_DIRECTORY,
+                "Folder": MEDIA_CLASS_DIRECTORY,
+                "BoxSet": MEDIA_CLASS_DIRECTORY,
+                "Episode": MEDIA_CLASS_EPISODE,
+                "Seaoson": MEDIA_CLASS_SEASON,
+            }
+            return BrowseMedia(
+                title=media["Name"],
+                media_content_id=media["Id"],
+                media_content_type=content_type_map.get(media["Type"], MEDIA_TYPE_NONE),
+                media_class=media_class_map.get(media["Type"], MEDIA_CLASS_DIRECTORY),
+                can_play=True,
+                can_expand=media["Type"] not in ["Movie", "Episode"],
```

Comment:
Use a separate list for this. The current selection is not complete, at least "Track" is missing (if you indeed intend to support music libraries).

Reply:
I'd like to support music libraries but I currently don't have any digital music to add to my instance to test with.
I should probably set one up and test, or avoid showing it at all...